### DISCUSSION TITLE
Bringing Plugin Support to CustomWidget

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11940] [Q&A]
Hi,
Currently we are trying to bring `Render Line Endings` Plugin support to our `CustomWidget`. We are rendering Monaco Editor directly within our `CustomWidget`. 

Once the plugin is installed Theia's `TextEditor` is able to pick up the extension and show accordingly.
![image](https://user-images.githubusercontent.com/80001447/204923534-85bf9b3a-60f6-4a9d-9315-cfeee1032bdf.png)
Whereas Our CustomWidget doesn't pick that up.
![image](https://user-images.githubusercontent.com/80001447/204923583-d04ec8ba-796a-4bfa-97f3-beda8948c505.png)

Can you provide insight on how Theia is able to achieve this and the place this happens such that we could try and bring this on to our CustomWidget.

Thanks

Additional Info: 
This is the said extension: https://open-vsx.org/extension/medo64/render-crlf

**Comment by maheshrajrp** [comment]
@msujew  @vince-fugnitto 
It would be great if you guys could help. Thanks.

**Reply by msujew** [reply]
I believe the reason why neither me nor Vince answer is because we both don't really have a good idea on how to accomplish this task. Everything inside of the plugin system is pretty much hardwired into using monaco editors directly. See [this code for example](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/editors-and-documents-main.ts). Maybe one approach could be to trick the framework into thinking that your custom widget is actually just a "normal" monaco editor. But no clue how this would work.

**Reply by maheshrajrp** [reply]
Thanks for replying. Will try this.
